mysql-test/suite/innodb/t/autoinc_import.test

```diff
@@ -0,0 +1,133 @@
+CREATE TABLE t10_1 LIKE t1;
+CREATE TABLE t10_1b LIKE t1b;
+ALTER TABLE t1 DISCARD TABLESPACE;
+ALTER TABLE t1b DISCARD TABLESPACE;
+ALTER TABLE t1t DISCARD TABLESPACE;
+ALTER TABLE t5_7 DISCARD TABLESPACE;
+ALTER TABLE t5_7b DISCARD TABLESPACE;
+ALTER TABLE t10_1 DISCARD TABLESPACE;
+ALTER TABLE t10_1b DISCARD TABLESPACE;
+FLUSH TABLES;
+
+# Update the PAGE_ROOT_AUTO_INC field of a few files.
+perl;
+do "$ENV{MTR_SUITE_DIR}/include/crc32.pl";
+sub update_autoinc
+{
+    my ($file, $value) = @_;
+    open(FILE, "+<$file") || die "Unable to open $file";
+    binmode FILE;
+    my $ps= $ENV{PAGE_SIZE};
+    my $page;
+    die "Unable to read $file" unless sysread(FILE, $page, $ps) == $ps;
+    my $full_crc32 = unpack("N",substr($page,54,4)) & 0x10; # FIL_SPACE_FLAGS
+    sysseek(FILE, 3*$ps, 0) || die "Unable to seek $file\n";
+    die "Unable to read $file" unless sysread(FILE, $page, $ps) == $ps;
+    substr($page,56,8)=pack("NN",0,$value);
+    my $polynomial = 0x82f63b78; # CRC-32C
+    if ($full_crc32) {
+        my $ck = mycrc32(substr($page, 0, $ps-4), 0, $polynomial);
+        substr($page, $ps-4, 4) = pack("N", $ck);
+    }
+    else
+    {
+        my $ck= pack("N",mycrc32(substr($page, 4, 22), 0, $polynomial) ^
+                     mycrc32(substr($page, 38, $ps - 38 - 8), 0, $polynomial));
+        substr($page,0,4)=$ck;
+        substr($page,$ps-8,4)=$ck;
+    }
+    sysseek(FILE, 3*$ps, 0) || die "Unable to rewind $file\n";
+    syswrite(FILE, $page, $ps)==$ps || die "Unable to write $file\n";
+    close(FILE) || die "Unable to close $file";
+}
+update_autoinc("$ENV{DATADIR}/test/t.ibd", 3);
+update_autoinc("$ENV{DATADIR}/test/tb.ibd", 346);
+update_autoinc("$ENV{DATADIR}/test/tt.ibd", 128);
+EOF
+
+--remove_file $DATADIR/test/t5_7.frm
+--remove_file $DATADIR/test/t5_7b.frm
+--copy_file $MYSQLTEST_VARDIR/std_data/autoinc_import_57.frm $DATADIR/test/t5_7.frm
+--copy_file $MYSQLTEST_VARDIR/std_data/autoinc_import_57.frm $DATADIR/test/t5_7b.frm
+--remove_file $DATADIR/test/t10_1.frm
+--remove_file $DATADIR/test/t10_1b.frm
+--copy_file $MYSQLTEST_VARDIR/std_data/autoinc_import_101.frm $DATADIR/test/t10_1.frm
+--copy_file $MYSQLTEST_VARDIR/std_data/autoinc_import_101.frm $DATADIR/test/t10_1b.frm
+--copy_file $DATADIR/test/t.ibd $DATADIR/test/t5_7.ibd
+--copy_file $DATADIR/test/tb.ibd $DATADIR/test/t5_7b.ibd
+--copy_file $DATADIR/test/t.ibd $DATADIR/test/t10_1.ibd
+--copy_file $DATADIR/test/tb.ibd $DATADIR/test/t10_1b.ibd
+--move_file $DATADIR/test/t.ibd $DATADIR/test/t1.ibd
+--move_file $DATADIR/test/tb.ibd $DATADIR/test/t1b.ibd
+--move_file $DATADIR/test/tt.ibd $DATADIR/test/t1t.ibd
+
+ALTER TABLE t1 IMPORT TABLESPACE;
+ALTER TABLE t1b IMPORT TABLESPACE;
+ALTER TABLE t1t IMPORT TABLESPACE;
+ALTER TABLE t5_7 IMPORT TABLESPACE;
+ALTER TABLE t5_7b IMPORT TABLESPACE;
+ALTER TABLE t10_1 IMPORT TABLESPACE;
+ALTER TABLE t10_1b IMPORT TABLESPACE;
+
+# Restart, so that the InnoDB tables will be loaded into the data dictionary.
```

Comment:
This is an important point. Does it mean that user could get into similar issues if the server is not restarted ?

1. If this is the case, I think we should try to deal with it, right ? May be by forcing a load.
2. We should have the mrt test that can check the behaviour before server restart.

Reply:
Normally, `ALTER TABLE…IMPORT TABLESPACE` should adjust the `AUTO_INCREMENT` value based on the contents of the `.cfg` file. Importing without a `.cfg` file is rather strongly discouraged. Starting with 89445b64fef2c3c9f2cfb9f572dd19b88f3a48df ([MDEV-26131](https://jira.mariadb.org/browse/MDEV-26131)) we only allow the `PRIMARY KEY` index to exist when importing without a `.cfg` file.